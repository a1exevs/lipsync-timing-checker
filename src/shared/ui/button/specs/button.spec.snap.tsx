import { render } from '@testing-library/react';
import { FileAudio } from 'lucide-react';
import React from 'react';

import Button from 'src/shared/ui/button/button';

describe('Button snapshots', () => {
  it('renders default', () => {
    const { asFragment } = render(<Button text="Button" onClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders primary variant', () => {
    const { asFragment } = render(<Button text="Primary" onClick={() => {}} variant="primary" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders secondary variant', () => {
    const { asFragment } = render(<Button text="Secondary" onClick={() => {}} variant="secondary" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders danger variant', () => {
    const { asFragment } = render(<Button text="Danger" onClick={() => {}} variant="danger" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders disabled', () => {
    const { asFragment } = render(<Button text="Disabled" onClick={() => {}} disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with icon', () => {
    const { asFragment } = render(<Button text="With Icon" onClick={() => {}} icon={<FileAudio />} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with icon on the right', () => {
    const { asFragment } = render(
      <Button text="Right Icon" onClick={() => {}} icon={<FileAudio />} iconPosition="right" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with widthFull', () => {
    const { asFragment } = render(<Button text="Full Width" onClick={() => {}} widthFull />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with additionalClasses', () => {
    const { asFragment } = render(<Button text="Additional" onClick={() => {}} additionalClasses="test-class" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
