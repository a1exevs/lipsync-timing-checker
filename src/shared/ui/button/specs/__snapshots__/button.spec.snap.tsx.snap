// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button snapshots renders danger variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-red-600 hover:bg-red-700 text-white"
    type="button"
  >
    <span>
      Danger
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders default 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white"
    type="button"
  >
    <span>
      Button
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders disabled 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white"
    disabled=""
    type="button"
  >
    <span>
      Disabled
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders primary variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white"
    type="button"
  >
    <span>
      Primary
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders secondary variant 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-gray-100 hover:bg-gray-200 text-gray-800 border border-gray-300"
    type="button"
  >
    <span>
      Secondary
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders with additionalClasses 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white test-class"
    type="button"
  >
    <span>
      Additional
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders with icon 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white"
    type="button"
  >
    <span>
      <svg
        aria-hidden="true"
        class="lucide lucide-file-audio"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.5 22h.5a2 2 0 0 0 2-2V7l-5-5H6a2 2 0 0 0-2 2v3"
        />
        <path
          d="M14 2v4a2 2 0 0 0 2 2h4"
        />
        <path
          d="M2 19a2 2 0 1 1 4 0v1a2 2 0 1 1-4 0v-4a6 6 0 0 1 12 0v4a2 2 0 1 1-4 0v-1a2 2 0 1 1 4 0"
        />
      </svg>
    </span>
    <span>
      With Icon
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders with icon on the right 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white flex-row-reverse"
    type="button"
  >
    <span>
      <svg
        aria-hidden="true"
        class="lucide lucide-file-audio"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.5 22h.5a2 2 0 0 0 2-2V7l-5-5H6a2 2 0 0 0-2 2v3"
        />
        <path
          d="M14 2v4a2 2 0 0 0 2 2h4"
        />
        <path
          d="M2 19a2 2 0 1 1 4 0v1a2 2 0 1 1-4 0v-4a6 6 0 0 1 12 0v4a2 2 0 1 1-4 0v-1a2 2 0 1 1 4 0"
        />
      </svg>
    </span>
    <span>
      Right Icon
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button snapshots renders with widthFull 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-full"
    type="button"
  >
    <span>
      Full Width
    </span>
  </button>
</DocumentFragment>
`;
